### PR TITLE
Implement Accounts base logic

### PR DESCRIPTION
**Pull request by Tojiddin20**
**Changes**:
- Added `Account` logic
- Removed `Method` due to needless
- Now we create a new `account` for user after `registration`
- We calculate `account balance` after each `transaction` 
- Renamed `Category` to `GeneralCategory` for better clarity
- Refactored `User Response` logic. Added `Extensions`

### CODE REVIEW

File: src/Controllers/AccountController.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Dtos.Account;
+using server.Models;
+using server.Responses.Account;
+
+namespace Server.Controllers;
+
+public class AccountController : ApiControllerBase
```
Comment (veyxov) [inline src/Controllers/AccountController.cs:10]: please use the new `Primary Constructor` feature of `dotnet 8` it looks much cleaner for DI

File: src/Controllers/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            return NotFound($"Account with id {id} not found!");
+        }
+
+        account.Name = updateAccountDto.Name;
+        account.Balance = updateAccountDto.Balance;
+        account.Currency = updateAccountDto.Currency;
+        account.Description = updateAccountDto.Description;
+        account.UpdatedDate = DateTime.Now;
+
+        _dbContext.Accounts.Update(account);
```
Comment (veyxov) [inline src/Controllers/AccountController.cs]: This is not needed, in this context the object is already a proxy and the changes are automatically tracked by EF

File: src/Controllers/AccountController.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        await _dbContext.SaveChangesAsync();
+
+        return Ok(new AccountResponse(account));
+    }
+
+    [HttpDelete("{id}")]
+    public async Task<ActionResult> Delete(int id)
+    {
+        Account? account = await _dbContext.Accounts.FindAsync(id);
```
Comment (veyxov) [inline src/Controllers/AccountController.cs]: `Find` is a little loose on `type safety`. I recommend using `FirstOrDefaultAsync(x => x.Id == id);`

File: src/Controllers/AccountController.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        if (account is null)
+        {
+            return NotFound($"Account with id {id} not found!");
+        }
+
+        _dbContext.Accounts.Remove(account);
```
Comment (veyxov) [inline src/Controllers/AccountController.cs]: Why do we need to get the object before deleting it?
You can just run `_dbContext.Accounts.Where(x => x.Id == id).ExecuteDeleteAsync()`

File: src/Controllers/AccountController.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            return NotFound($"Account with id {id} not found!");
+        }
+
+        return Ok(new AccountResponse(account));
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<List<AccountResponse>>> GetUserAccounts(int userId)
+    {
+        List<Account> accounts = await _dbContext.Accounts.Where(account => account.UserId == userId).ToListAsync();
```
Comment (veyxov) [inline src/Controllers/AccountController.cs]: You're getting all the fields from the database and removing the unnecessary ones in the memory. This is not needed, you can *only* get the needed fields right from the database, like:

`return await _context.Accounts.Where(...).Select(x => MapToResponse(x).ToListAsync()`

File: src/Controllers/AuthController.cs [hunk 6]
```diff
@@ -40,6 +40,20 @@ public async Task<ActionResult<UserResponse>> Register(CreateUserDto createUserD
         _dbContext.Users.Add(user);
         await _dbContext.SaveChangesAsync();
 
+        // Todo: Extract into class
```
Comment (veyxov) [inline src/Controllers/AuthController.cs]: This should be an event, when a user is created an account should be created like:
`UserCreationEvent -> CheckForDefaultAcount`.

File: src/Controllers/ExpenseController.cs [hunk 7]
```diff
@@ -19,6 +19,17 @@ public ExpenseController(ApplicationDbContext dbContext)
...
+        int accountId = createExpenseDto.AccountId;
+        // Todo: Extract into class
+        Account? account = await _dbContext.Accounts.FindAsync(accountId);
+        if (account is null)
+        {
+            return NotFound($"Account by id {accountId} not found!");
+        }
+
+        account.Balance -= createExpenseDto.Amount;
+        _dbContext.Accounts.Update(account);
```
Comment (veyxov) [inline src/Controllers/ExpenseController.cs]: Why?

File: src/Dtos/Account/CreateAccountDto.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+public class CreateAccountDto
+{
+    public required string Name { get; set; } 
+
+    public required decimal Balance { get; set; } 
+
+    public required string Currency { get; set; }
+
+    public string? Description { get; set; } = null;
```
Comment (veyxov) [inline src/Dtos/Account/CreateAccountDto.cs]: null is the default value for `string?` no need to explicitly set it

File: src/Controllers/AccountController.cs (hunk 1 above)
Comment (Tojiddin20) [reply src/Controllers/AccountController.cs:10]: I will upgrade `dotnet` to `8` in other `PR`

**Review by ifteakhor — APPROVED** [review]

**Pull request merged**